src/devices/RGBLedMatrix/samples/RGBLedMatrix.sample.cs

```diff
@@ -0,0 +1,678 @@
+            float c1 = c.X + 6.0f;
+
+            Vector3 v1 = Add(new Vector3(0.0f, 4.0f, 2.0f), c.X * 6.0f);
+            Vector3 rgb = clamp(Add(abs(Add(mod(v1, 6.0f), -3.0f)), -1.0f), 0.0f, 1.0f);
+
+            rgb = rgb*rgb*(Add(-2.0f * rgb, 3.0f)); // cubic smoothing
+
+            return c.Z * mix(new Vector3(1.0f, 1.0f, 1.0f), rgb, c.Y);
+        }
+
+        private static float smoothstep(float edge0, float edge1, float x)
+        {
+            // Scale, bias and saturate x to 0..1 range
+            x = Math.Clamp((x - edge0) / (edge1 - edge0), 0.0f, 1.0f);
+            // Evaluate polynomial
+            return x * x * (3 - 2 * x);
+        }
+
+        private static Vector3 Star(Vector2 uv, float time)
+        {
+            Vector2 p = uv - new Vector2(0.5f, 0.5f);
+            float a = (float)(Math.Atan2(p.Y, p.X) / 2f / Math.PI);
+
+            a = mod(a + time / 10.0f, 1.0f);
+
+            float r = p.Length();
+            float n = 5.0f;
+
+            float s1 = 0.2f;
+            float s2 = 0.5f;
+            float srange = s2 - s1;
+            float x = Math.Abs(mod(a, 1.0f / n) * n - 0.5f) * srange / 0.5f + s1;
+
+            float blur = 4.0f + 3.0f * (float)Math.Sin(time * 2f * Math.PI / 5f);
+            float c = smoothstep(0.0f, 1.0f, (x - r) * blur);//x <= r ? 0.0 : 1.0;
+            //// return hsv2rgb_smooth(new Vector3(mod(time / 11.0f, 1.0f), 1.0f, c));
+
+            float ha = a;
+            //ha = ha + time / 10f;
+            float h = mod(ha, 1.0f);
+            //h = (float)Math.Floor(h * colFac) / colFac;
+            //h = mod(h + mod(time / 4.0f, 0.5f), 1.0f);
+
+            float v = c;
+            //v = (float)Math.Floor(v * colFac2) / colFac2;
+
+            float s = 1.0f;
+            //s = (float)Math.Pow(r * 5.0f, 0.5f);
+            return hsv2rgb_smooth(new Vector3(h, s, v));
+        }
+
+        private static Vector3 HSV(Vector2 uv, float time)
+        {
+            Vector2 p = uv - new Vector2(0.5f, 0.5f);
+            float a = (float)(Math.Atan2(p.Y, p.X) / 2f / Math.PI);
+
+            a = mod(a + time / 10.0f, 1.0f);
+
+            float r = p.Length();
+
+            float ha = a;
+            //ha = ha + time / 10f;
+            float h = mod(ha, 1.0f);
+            //v = (float)Math.Floor(v * colFac2) / colFac2;
+
+            float s = 1.0f;
+            float v = r * 2.0f;
+
+            //s = (float)Math.Pow(r * 5.0f, 0.5f);
+            return hsv2rgb_smooth(new Vector3(a, s, v));
+        }
+
+        private static byte Col(float x)
+        {
+            x *= 255f;
+            x = Math.Clamp(x, 0f, 255f);
+            return (byte)x;
+        }
+
+        private static byte Col(double x, double d, double e)
+        {
+            x *= e;
+            x = Math.Pow(x, d);
+            x = Math.Clamp(x, 0.0f, 1.0f);
+            return (byte)(x * 255);
+        }
+
+        private static byte ColR(double x)
+        {
+            return Col(x, 1.9, 0.95);
```

Comment:
I think we might be doing gamma twice (here and the RGB matrix driver)

Reply:
This is still producing a good result. I'll leave it for now as we may do more enhancement in general for the gamma in the driver itself. I think we'll need to support brightness too.